biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts

```diff
@@ -1,28 +1,72 @@
 import * as jsgraphs from 'js-graph-algorithms';
 import {
   StateGraphVertex,
-  NewStateGraphVertex,
   StateGraph,
   StateGraphEdge,
 } from '@biogrid/grid-simulator';
+import { GridItem } from 'libs/grid-simulator/src/lib/grid-item';
 
 export class BiogridState implements StateGraph {
   graph: jsgraphs.WeightedDiGraph;
   graphIndexToVertex: StateGraphVertex[];
 
-  constructor(numberOfVertices: number, vertices: NewStateGraphVertex[]) {
-    this.graph = new jsgraphs.WeightedDiGraph(numberOfVertices);
-    this.graphIndexToVertex = new Array<StateGraphVertex>(numberOfVertices);
+  constructor(vertices: StateGraphVertex[]) {
+    this.graph = new jsgraphs.WeightedDiGraph(vertices.length);
+    this.graphIndexToVertex = new Array<StateGraphVertex>(vertices.length);
     vertices.forEach((vertex, i) => {
-      this.graphIndexToVertex[i] = vertex.vertex;
-      this.addAllEdges(i, vertex.edges);
+      this.graphIndexToVertex[i] = vertex;
+    });
+    vertices.forEach((vertex, i) => {
+      this.addCompletlyConnectedVertex(i, vertex);
     });
   }
 
   public getGraph() {
     return this.graph;
   }
 
+  public convertStateGraphToMST() {
+    this.graph = this.getMinimumSpanningTree(this.graph);
+  }
+
+  private addCompletlyConnectedVertex(
+    newVertexIndex: number,
+    vertex: StateGraphVertex
+  ) {
+    const edges: StateGraphEdge[] = [];
+    for (let i = 0; i < this.graph.V; i++) {
+      if (newVertexIndex !== i) {
+        edges.push({
+          toIndex: i,
+          weight: this.calculateDistance(vertex, this.graphIndexToVertex[i]),
+          label: `${newVertexIndex}-to-${i}`,
+        });
+      }
+    }
+    this.addAllEdges(newVertexIndex, edges);
+  }
+
+  /**
+   * Calculate distance via Pythagorean's theorem
+   */
+  private calculateDistance(v1: StateGraphVertex, v2: StateGraphVertex) {
+    return Math.sqrt(
+      Math.pow(v1.getPosition().x - v2.getPosition().x, 2) +
+        Math.pow(v1.getPosition().y - v2.getPosition().y, 2)
+    );
+  }
```

Comment:
So for instance here, on the question of if the positions are allowed to take in negative. If they are not taking in negatives, then their distances apart should be recalculated.
Ignore if they are taking in negatives